### PR TITLE
Upgrade kubebuilder to 0.1.12.

### PR DESCRIPTION
**Pull request by gyliu513**
Fixed https://github.com/kubernetes/cluster-registry/issues/243

/cc @font @madhusudancs @perotinus @pmorie

/sig multicluster

### CODE REVIEW

File: pkg/apis/clusterregistry/v1alpha1/zz_generated.kubebuilder.go [hunk 1]
```diff
@@ -86,72 +86,68 @@ var (
 				OpenAPIV3Schema: &v1beta1.JSONSchemaProps{
 					Type: "object",
 					Properties: map[string]v1beta1.JSONSchemaProps{
-						"apiVersion": {
+						"apiVersion": v1beta1.JSONSchemaProps{
```
Comment (perotinus) [inline pkg/apis/clusterregistry/v1alpha1/zz_generated.kubebuilder.go]: I think you need to run `gofmt -s` against this file.

File: hack/imports.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/*
+Copyright YEAR The Kubernetes Authors.
```
Comment (perotinus) [inline hack/imports.go]: Can you set the year?

**Comment by perotinus** [comment]
@gyliu513 Apologies for letting this sit for so long! This looks reasonable to me overall. Was there a different version of the docs dependencies pulled in by kubebuilder? I noticed a significant amount of churn there which I didn't look at in detail.

File: pkg/apis/clusterregistry/v1alpha1/zz_generated.kubebuilder.go (hunk 1 above)
Comment (gyliu513) [reply pkg/apis/clusterregistry/v1alpha1/zz_generated.kubebuilder.go]: sure, will fix this.

File: hack/imports.go (hunk 2 above)
Comment (gyliu513) [reply hack/imports.go]: ok

**Comment by gyliu513** [comment]
@perotinus I will check more for this, but for now, the 0.1.12 has just released, so I'd prefer that we upgrade to 0.1.12 directly in this PR, what do you think?

File: pkg/apis/clusterregistry/v1alpha1/zz_generated.kubebuilder.go (hunk 1 above)
Comment (gyliu513) [reply pkg/apis/clusterregistry/v1alpha1/zz_generated.kubebuilder.go]: @perotinus this file is generated automatically, I think we do not need to update this?

Comment (gyliu513) [reply pkg/apis/clusterregistry/v1alpha1/zz_generated.kubebuilder.go]: Done

**Comment by perotinus** [comment]
@gyliu513 Thank you! Upgrading to 0.1.12 seems like the right approach to me as well.

**Comment by perotinus** [comment]
Can you adjust the comment message on the first commit, https://github.com/kubernetes/cluster-registry/pull/244/commits/ff31d318b867176bbd211e3e07148e522b204f58? Other than that, LGTM.

**Comment by gyliu513** [comment]
@perotinus the first pr is updating to 0.11, and the second pr is updating to 0.12 at https://github.com/kubernetes/cluster-registry/pull/244/commits/c3a4a4ae352d8d678f06b22a3c2ecf5961f50024 

Can we keep the commit message as it is as this can reflect the work process of this PR?

**Comment by perotinus** [comment]
@gyliu513 Sure, that seems fine to me. In that case, can you update the commit message of https://github.com/kubernetes/cluster-registry/commit/c3a4a4ae352d8d678f06b22a3c2ecf5961f50024 so that it specifies that it's updating to 0.12?

**Comment by gyliu513** [comment]
@perotinus the commit message does include 0.1.12 https://github.com/kubernetes/cluster-registry/commit/c3a4a4ae352d8d678f06b22a3c2ecf5961f50024

image

**Comment by perotinus** [comment]
@gyliu513 Ah, OK, sorry about that.

**Pull request merged**